src/Aws/Lambda/Meta/Dispatch.hs

```diff
@@ -9,32 +8,32 @@ module Aws.Lambda.Meta.Dispatch
   , Runtime.LambdaResult(..)
   ) where
 
+import qualified Data.Char as Char
 import Data.Function ((&))
 import Data.Text (Text)
 import qualified Data.Text as Text
-import qualified Data.Char as Char
 
 import Data.Aeson
 import qualified Data.ByteString.Lazy.Char8 as LazyByteString
 import qualified Language.Haskell.TH as Meta
 
 import Aws.Lambda.Meta.Common
+import qualified Aws.Lambda.Meta.Main as Main
+import qualified Aws.Lambda.Runtime.ApiGatewayInfo as ApiGatewayInfo
 import qualified Aws.Lambda.Runtime.Common as Runtime
 import qualified Aws.Lambda.Runtime.Error as Error
-import qualified Aws.Lambda.Runtime.ApiGatewayInfo as ApiGatewayInfo
 import qualified Control.Exception as Unchecked
-import qualified Aws.Lambda.Meta.Main as Main
-import Data.Typeable (Typeable, typeRep, Proxy (..))
+import Data.Typeable (Proxy (..), Typeable, typeRep)
 
 {-| Helper function that the dispatcher will use to
 decode the JSON that comes as an AWS Lambda event into the
 appropriate type expected by the handler.
 -}
-decodeObj :: forall a. (FromJSON a, Typeable a) => String -> Either Error.Parsing a
+decodeObj :: forall a. (FromJSON a, Typeable a) => LazyByteString.ByteString -> Either Error.Parsing a
```

Comment:
The packing and unpacking seemed redundant, so I just made it use ByteString instead.